### PR TITLE
Coverage docker & kokoro

### PR DESCRIPTION
**Pull request by antonrohr**
The docker container is already pushed to the cloud project. The successful test is here:

https://fusion.corp.google.com/runanalysis/info/prod:orbitprofiler%2Fbuilds%2Flinux%2Fclang7_relwithdebinfo%2Fcontinuous/prod:orbitprofiler%2Fbuilds%2Flinux%2Fclang7_relwithdebinfo%2Fcontinuous/KOKORO/a888eaee-c02f-44c5-9039-c75b10fa0855/1613132417885/build%20%23709/Targets

Bucket:
https://pantheon.corp.google.com/storage/browser/orbitprofiler-build/gcp_ubuntu_artifacts/prod/orbitprofiler/builds/linux/clang7_relwithdebinfo/continuous/709/20210212-042019;tab=objects?prefix=&forceOnObjectsSortingFiltering=false

(See coverage sub folder)

### CODE REVIEW

File: kokoro/builds/build.sh [hunk 1]
```diff
@@ -142,6 +142,13 @@ if [ -n "$1" ]; then
...
   conan build -bf "${REPO_ROOT}/build/" "${REPO_ROOT}"
 
+  if [[ $CONAN_PROFILE == "coverage_clang9" ]]; then 
+    echo "Starting to generate unit test coverage report"
+    mkdir -p "${REPO_ROOT}/build/package/coverage"
+    ${REPO_ROOT}/contrib/unit_test_coverage/generate_coverage_report.sh \
+            "${REPO_ROOT}/src" "${REPO_ROOT}/build" "${REPO_ROOT}/build/package/coverage"
+  fi
+
   if [[ $CONAN_PROFILE != "iwyu" ]]; then
```
Comment (beckerhe) [inline kokoro/builds/build.sh:152]: I think we don't want to call `conan package` when using the `coverage_clang9`, just like include-what-you-use.

If we did, we would store all the binaries in the bucket, again. There is another job doing that already.

Comment (beckerhe) [reply kokoro/builds/build.sh:152]: The same applies to the ZIP file generation in line 206. This has to be exluded as well.

Comment (beckerhe) [reply kokoro/builds/build.sh:152]: One more thing. Could you put this part further down in the script? Maybe after the `Analyze include-what-you-use` results?

**Review by beckerhe — COMMENTED** [review]
Very nice. Thanks a lot, Anton!

I added some minor comments regarding the build.sh script. The rest looks good!

**Comment by beckerhe** [comment]
One more thing: You might wanna add the kokoro build config to this PR. (Could also be a separate PR)

File: kokoro/builds/build.sh (hunk 1 above)
Comment (antonrohr) [reply kokoro/builds/build.sh]: I fixed this and tested that it works.

**Comment by antonrohr** [comment]
> One more thing: You might wanna add the kokoro build config to this PR. (Could also be a separate PR)

Came to the same conclusion while looking at the kokoro scripts. I added them here. ptal

**Review by beckerhe — APPROVED** [review]
Nice. LGTM! Thanks!

**Pull request merged**